Confusing if statement in AIWarrush.cpp

Around [line 148 of src/AIWarrush.cpp](https://github.com/Globulation2/glob2/blob/7c23bf874c1ae5263dc2cff552864821e2a526e4/src/AIWarrush.cpp#L148), there is a confusing if statement. It gives several warnings when compiled with GCC 12.2 on MSys2 Mingw 64-bit, one for each line. Maybe the condition was supposed to check that the building's type was not a building that doesn't need workers. The types of building it checks (hospital, inn, swimming pool, barracks, school) don't need workers unless the building is currently under construction or being upgraded.
```
src/AIWarrush.cpp: In member function 'Building* AIWarrush::getBuildingWithoutWorkersAssigned(Sint32, int) const':
src/AIWarrush.cpp:152:83: warning: comparison of constant 'IntBuildingType::ATTACK_BUILDING' with boolean expression is always true [-Wbool-compare]
  152 |                                         || ((b->type->shortTypeNum==shortTypeNum) != (IntBuildingType::ATTACK_BUILDING)
      |                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/AIWarrush.cpp:153:90: warning: comparison of constant 'IntBuildingType::HEAL_BUILDING' with boolean expression is always true [-Wbool-compare]
  153 |                                                 && (b->type->shortTypeNum==shortTypeNum) != (IntBuildingType::HEAL_BUILDING)
      |                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/AIWarrush.cpp:154:90: warning: comparison of constant 'IntBuildingType::WALKSPEED_BUILDING' with boolean expression is always true [-Wbool-compare]
  154 |                                                 && (b->type->shortTypeNum==shortTypeNum) != (IntBuildingType::WALKSPEED_BUILDING)
      |                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/AIWarrush.cpp:155:90: warning: comparison of constant 'IntBuildingType::SWIMSPEED_BUILDING' with boolean expression is always true [-Wbool-compare]
  155 |                                                 && (b->type->shortTypeNum==shortTypeNum) != (IntBuildingType::SWIMSPEED_BUILDING)
      |                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/AIWarrush.cpp:156:90: warning: comparison of constant 'IntBuildingType::SCIENCE_BUILDING' with boolean expression is always true [-Wbool-compare]
  156 |                                                 && (b->type->shortTypeNum==shortTypeNum) != (IntBuildingType::SCIENCE_BUILDING)
      |                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/AIWarrush.cpp
+++ b/src/AIWarrush.cpp
@@ -144,16 +144,15 @@ Building *AIWarrush::getBuildingWithoutWorkersAssigned(Sint32 shortTypeNum, int 
 	for (int i=0; i<Building::MAX_COUNT; i++)
 	{
 		Building *b=myBuildings[i];
-		//TODO: b->type->shortTypeNum==shortTypeNum != IntBuildingType::HEAL_BUILDING looks fishy not only to g++
 		if (	(b != NULL)
 				&& (b->type->shortTypeNum == shortTypeNum)
 				&& (b->maxUnitWorking != num_workers)
 				&& (b->constructionResultState != Building::NO_CONSTRUCTION
-					|| ((b->type->shortTypeNum==shortTypeNum) != (IntBuildingType::ATTACK_BUILDING)
-						&& (b->type->shortTypeNum==shortTypeNum) != (IntBuildingType::HEAL_BUILDING)
-						&& (b->type->shortTypeNum==shortTypeNum) != (IntBuildingType::WALKSPEED_BUILDING)
-						&& (b->type->shortTypeNum==shortTypeNum) != (IntBuildingType::SWIMSPEED_BUILDING)
-						&& (b->type->shortTypeNum==shortTypeNum) != (IntBuildingType::SCIENCE_BUILDING)
+					|| ((shortTypeNum != IntBuildingType::ATTACK_BUILDING)
+						&& (shortTypeNum != IntBuildingType::HEAL_BUILDING)
+						&& (shortTypeNum != IntBuildingType::WALKSPEED_BUILDING)
+						&& (shortTypeNum != IntBuildingType::SWIMSPEED_BUILDING)
+						&& (shortTypeNum != IntBuildingType::SCIENCE_BUILDING)
 			)))
 		{
 			return b;

